xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -276,9 +276,11 @@ func (s) TestErrorFromParentLB_ResourceNotFound(t *testing.T) {
 		t.Fatalf("RPCs did not fail after removal of Cluster resource")
 	}
 
-	// Ensure that the ClientConn is in TransientFailure.
-	if state := cc.GetState(); state != connectivity.TransientFailure {
-		t.Fatalf("Unexpected connectivity state for ClientConn, got: %s, want %s", state, connectivity.TransientFailure)
+	// Ensure that the ClientConn moves to TransientFailure.
+	for state := cc.GetState(); state != connectivity.TransientFailure; state = cc.GetState() {
```

Comment:
idk how these instructions work exactly. But if they are executed left to right, what happens if state != connectivity.TransientFailure, yield, other thread causes Client Conn to get a TF update, and then state = cc.GetState() occurs, reads TF, and 281 causes a context expiration due to it already being in TF and never leaving TF?

Reply:
This is `for init; condition; post { }`. So, the `post` only runs after the block of code within the `for` loop, but before the next time `condition` is run.

See: https://go.dev/doc/effective_go#for

Reply:
Ah ok, thanks for explanation. I actually didn't know that the post runs after the for, but thinking about it a common scenario is i := 0; i < 5; i++, and when I use i in the for it's at the index.